hugo/assets/scripts/ast/statemachine.ts

```diff
@@ -0,0 +1,145 @@
+import { type } from "os";
+import { AstNode, LangiumAST } from "./ast-tools";
+
+
+
+export class StateMachineTools {
+    currentState: StateMachineState;
+    ast: StateMachineAstNode;   
+    constructor(ast: StateMachineAstNode) {
+        this.ast = ast;
+        this.currentState = this.getInitialState(ast);
+    }
+
+    /**
+     * get the initial state from the AST
+     * @param ast The AST to get the initial state from
+     * @returns 
+     */
+    private getInitialState(ast: StateMachineAstNode): StateMachineState {
```

Comment:
Can probably drop the parameter, and instead reference `this.ast`, as it's set in the constructor above.